### DISCUSSION TITLE
There will be delay on the jump page on solw 3g

### DISCUSSION THREAD

**Discussion by 19Qingfeng** [remix-run/remix#9117] [General]
hi, I recently used remix on my mobile site. Although this is the expected behavior, I still want to explore whether this kind of behavior can be optimized.

If there is a loader in the next route when I use usenavigate during SPA redirection. In slow 3g, it is necessary to wait for the server to respond (even if it is already delayed), which will cause a slight lag when clicking.

I am not sure if Remix will have any optimized behavior for this behavior in the future. In theory, when switching to SPA, I believe that it should not rely on the server's response (the loader should be executed on the client side).

However, due to the current design philosophy of Remix, the loader will not be built on the client side. So I was wondering if there is another way, such as preloading the loader of the next route before the user clicks to jump, to reduce waiting time when clicking?

I know that usenavigate(). state can be used to load cutscenes, but I want to minimize the occurrence of cutscenes as much as possible (or even avoid them), so I would like to seek optimization for preloading or other ways to address this issue.

**Comment by lili21** [comment]
There are several options you can try,

### Prefetch

I would recommend you to use `<Link />` instead of `useNavigate` whenever posible. Cause you can prefetch the next route by using the `prefetch` prop.

Also you can try the `PrefetchPageLinks` component for prefetching. 

## clientLoader

You can use clientLoader if you don't want the loader block the navigation.

**Reply by 19Qingfeng** [reply]
Okay, thank you. The only regret is that we are currently using version 1.15 without clientLoader.
I will use PrefetchPageLinks to handle my pages, but the freshness of the loaderData data is indeed a headache.
